src/makeInfinityAdder.js

```diff
@@ -6,7 +6,22 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  const callback = (num) => {
+    if (num === undefined) {
+      const oldSum = sum;
+
+      sum = 0;
+
+      return oldSum;
+    } else {
+      sum += num;
+
+      return callback;
+    }
+  };
+
+  return callback;
```

Comment:
```suggestion
    }
      sum += num;

      return callback;
  };
```
You can get rid of `else` and it will mean that `sum += num` will execute every time when `num !== undefined` (`callback` will be returning as well)